### PR TITLE
Add support for Pulsar code editor on Windows

### PR DESCRIPTION
**Pull request by confused-Techie**
## Description

Adds support for the [Pulsar](https://pulsar-edit.dev) code editor on Windows. A follow-up to #16220 

### Screenshots

image

## Release notes

- [Added] Add support for Pulsar code editor on Windows

Notes:

Apologies if my commit history is a little messy. But hopefully it's not an issue, thanks a ton to anyone who is able to take a look at this!

### CODE REVIEW

**Review by sergiou87 — DISMISSED** [review]
Works as expected, thank you for your contribution! 💖

File: app/src/lib/editors/win32.ts [hunk 1]
```diff
@@ -479,6 +479,15 @@ const editors: WindowsExternalEditor[] = [
     displayNamePrefix: 'DataSpell ',
     publishers: ['JetBrains s.r.o.'],
   },
+  {
+    name: 'Pulsar',
+    registryKeys: [
+      CurrentUserUninstallKey('0949b555-c22c-56b7-873a-a960bdefa81f')
```
Comment (sergiou87) [inline app/src/lib/editors/win32.ts]: Let's please the linter!
```suggestion
      CurrentUserUninstallKey('0949b555-c22c-56b7-873a-a960bdefa81f'),
```

**Comment by sergiou87** [comment]
@confused-Techie could you please ~~rebase~~ update this PR with the latest `development`? #17078 changed the `displayNamePrefix` attribute to `displayNamePrefixes`, so I think this PR needs to adapt to that… sorry! 😅

**Comment by confused-Techie** [comment]
> @confused-Techie could you please ~rebase~ update this PR with the latest `development`? #17078 changed the `displayNamePrefix` attribute to `displayNamePrefixes`, so I think this PR needs to adapt to that… sorry! 😅

Absolutely, I'll be able to get to this later today. Thanks a ton for reviewing my PR!

**Comment by confused-Techie** [comment]
@sergiou87 as requested, I've updated this PR to reflect the new changes upstream, and merged all changes upstream here, to hopefully ensures tests pass as expected.

Again thanks a ton for spending your time on this one

File: app/src/lib/editors/win32.ts [hunk 2]
```diff
@@ -488,6 +488,15 @@ const editors: WindowsExternalEditor[] = [
     displayNamePrefixes: ['DataSpell '],
     publishers: ['JetBrains s.r.o.'],
   },
+  {
+    name: 'Pulsar',
+    registryKeys: [
+      CurrentUserUninstallKey('0949b555-c22c-56b7-873a-a960bdefa81f'),
+    ],
+    executableShimPaths: [['..', 'pulsar', 'Pulsar.exe']],
+    displayNamePrefixs: ['Pulsar'],
```
Comment (sergiou87) [inline app/src/lib/editors/win32.ts]: Typo:
```suggestion
    displayNamePrefixes: ['Pulsar'],
```

Comment (confused-Techie) [reply app/src/lib/editors/win32.ts]: Whoops, thanks for resolving that

**Review by sergiou87 — APPROVED** [review]

**Comment by Meisamrzp2** [comment]
ok thanks

**Pull request merged**